### DISCUSSION TITLE
"Target is already being recorded" warning with custom agent

### DISCUSSION THREAD

**Discussion by kotcrab** [NationalSecurityAgency/ghidra#3559] [Debugger]
I made my own debugger agent which connects to the PPSSPP emulator. Every feature seems to be working though there's one thing I don't understand. When the `Process` transitions between `RUNNING` and `STOPPED` states this warning is always logged:
```
Target is already being recorded: <PpssppModelTargetProcess: path=Process model=allegrex.agent.ppsspp.model.PpssppDebuggerObjectModel@210e1a6 schema=Process>
```
The emulator only allows to interrupt/resume the whole process and not individual threads so in my code only the `Process` implements `TargetExecutionStateful`.

The code that triggers this warning is [here](https://github.com/NationalSecurityAgency/ghidra/blob/c19276091f274a9ef0850c904c743f61c850854e/Ghidra/Debug/Debugger/src/main/java/ghidra/app/plugin/core/debug/gui/objects/DebuggerObjectsProvider.java#L1715-L1721). Looks like the only way to avoid it is to uncheck "Record Automatically" but is that how it's supposed to work?

**Comment by d-millar** [comment]
@kotcrab OK, first off, let me say I'm impressed!  Writing your own agent is not for the faint of heart.  You rock.

WRT your question, that is not how that's supposed to work, but, looking at the Linux agent, it's doing the same thing.  Think we screwed something up somewhere.  Am putting in a ticket now.  Thanks for pointing this out.  I suppose as an extremely lame workaround you can uncheck "Record Automatically" and hit "Record" after you've launched, but you shouldn't have to do that.

**Reply by Nemoumbra** [reply]
Any news on that problem? Maybe it'd be worth it making an issue from this discussion?

**Reply by kotcrab** [reply]
This might have gotten fixed already, I can't seem to reproduce it.

**Reply by d-millar** [reply]
@Nemoumbra are you still seeing this and, if so, which version of Ghidra?

**Reply by Nemoumbra** [reply]
No, I don't see anything right now.

**Reply by d-millar** [reply]
ok, will assume it’s not an issue for the moment, but, if it recurs, please give us a shout.
